crates/sncast/tests/integration/data_transformer.rs

```diff
@@ -0,0 +1,1022 @@
+
+    Ok(())
+}
+
+#[tokio::test]
+async fn test_happy_case_simple_function_serialized_input() -> anyhow::Result<()> {
+    let contract_class = CLASS.get_or_init(init_class).await.to_owned();
+
+    let input = vec![String::from("0x64")];
+
+    let result = transform(
+        &input,
+        contract_class,
+        &get_selector_from_name("simple_fn").unwrap(),
+    )?;
+
+    let expected_output: Vec<Felt> = vec![100.into()];
+
+    assert_eq!(result, expected_output);
+
+    Ok(())
+}
+
+#[tokio::test]
+async fn test_happy_case_u256_function_cairo_expressions_input() -> anyhow::Result<()> {
+    let contract_class = CLASS.get_or_init(init_class).await.to_owned();
+
+    let input = vec![U256::MAX.to_string()];
+
+    let result = transform(
+        &input,
+        contract_class,
+        &get_selector_from_name("u256_fn").unwrap(),
+    )?;
+
+    let expected_output = vec![
+        Felt::from_hex_unchecked("0xffffffffffffffffffffffffffffffff"),
+        Felt::from_hex_unchecked("0xffffffffffffffffffffffffffffffff"),
+    ];
+
+    assert_eq!(result, expected_output);
+
+    Ok(())
+}
+
+#[tokio::test]
+async fn test_happy_case_u256_function_serialized_input() -> anyhow::Result<()> {
+    let contract_class = CLASS.get_or_init(init_class).await.to_owned();
+
+    let input = vec![String::from("0x2137"), String::from("0x0")];
+
+    let result = transform(
+        &input,
+        contract_class,
+        &get_selector_from_name("u256_fn").unwrap(),
+    )?;
+
+    let expected_output = vec![
+        Felt::from_hex_unchecked("0x2137"),
+        Felt::from_hex_unchecked("0x0"),
+    ];
+
+    assert_eq!(result, expected_output);
+
+    Ok(())
+}
+
+#[tokio::test]
+async fn test_happy_case_signed_function_cairo_expressions_input() -> anyhow::Result<()> {
+    let contract_class = CLASS.get_or_init(init_class).await.to_owned();
+
+    let input = vec![String::from("-273")];
+
+    let result = transform(
+        &input,
+        contract_class,
+        &get_selector_from_name("signed_fn").unwrap(),
+    )?;
+
+    let expected_output = vec![Felt::from(-273i16)];
+
+    assert_eq!(result, expected_output);
+
+    Ok(())
+}
+
+#[tokio::test]
+async fn test_happy_case_signed_function_serialized_input() -> anyhow::Result<()> {
+    let contract_class = CLASS.get_or_init(init_class).await.to_owned();
+
+    let input = vec![Felt::from(-273i16).to_hex_string()];
+
+    let result = transform(
+        &input,
+        contract_class,
+        &get_selector_from_name("signed_fn").unwrap(),
+    )?;
+
+    let expected_output = vec![Felt::from(-273i16)];
+
+    assert_eq!(result, expected_output);
+
+    Ok(())
+}
+
+// Problem: Although transformer fails to process the given input as `i32`, itthen succeeds to interpret it as `felt252`
+// Overflow checks will not work for functions having the same serialized and Cairo-like calldata length.
+// User must provide a type suffix or get the invoke-time error
```

Comment:
We should probably document that and create an issue to fix it in the future.

Reply:
Or we could just say that this is intended, but it's a bit weird to me.

Reply:
Sadly we can't solve this since we allow passing both Cairo-like and serialized calldata and don't differentiate them before transformation

Reply:
This is a bit unfortunate but I think we can make do with documenting it. I'd consider creating an issue and mentioning it here though if we ever decide to resolve this.

Reply:
Added an issue - [#2559](https://github.com/foundry-rs/starknet-foundry/issues/2559#issue-2567176737)